### PR TITLE
improve rest api documentation: first draft

### PR DESCRIPTION
**Pull request by unixpi**
improvements to rest api page modulo fancy javascript

### CODE REVIEW

File: docs/the_nimbus_book/src/rest-api.md [hunk 1]
```diff
@@ -1,31 +1,172 @@
 # REST API
 
-Nimbus exposes a high-performance implementation of the [Beacon Node API](https://ethereum.github.io/beacon-APIs/).
+Nimbus exposes an **extremely fast** and high-performance implementation of the standard [Beacon Node API](https://ethereum.github.io/beacon-APIs/). The goal of a standard API is to improve interoperability between various beacon node implementations.
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/rest-api.md]:
```suggestion
Nimbus exposes an **extremely fast** and high-performance implementation of the standard [Beacon Node API](https://ethereum.github.io/beacon-APIs/). The API allows you to use Nimbus together with third-party tooling such as validator clients, block explorers as well as your own monitoring infrastructure.

File: docs/the_nimbus_book/src/rest-api.md [hunk 2]
```diff
@@ -1,31 +1,172 @@
...
 
-The API is a `REST` interface accessed via `HTTP`. The API should not, unless protected by additional security layers, be exposed to the public Internet as the API includes multiple endpoints which could open your node to denial-of-service (DoS) attacks.
+The API is a `REST` interface accessed via `HTTP`. **The API should not**, unless protected by additional security layers, **be exposed to the public Internet** as the API includes multiple endpoints which could open your node to denial-of-service (DoS) attacks.
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/rest-api.md]:
```suggestion
The API is a `REST` interface accessed via `HTTP`. **The API should not be exposed to the public Internet** unless protected by additional security: it includes multiple endpoints which could open your node to denial-of-service (DoS) attacks.
```

File: docs/the_nimbus_book/src/rest-api.md [hunk 3]
```diff
@@ -1,31 +1,172 @@
...
+```
+curl -X GET http://unstable.mainnet.beacon-api.nimbus.team/eth/v1/node/version
+```
+
+#### Prater unstable branch
+```
+curl -X GET  http://unstable.prater.beacon-api.nimbus.team/eth/v1/node/version
+```
+
+Note that right now these are very much unstable testing instances. They may be unresponsive at times - so **please do not rely on them for validation**. We may also disable them at any time without warning.
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/rest-api.md]:
```suggestion
The test endpoints are part of pre-release testing and run an unstable version of Nimbus - we welcome reports about any problems you might have with them.

They may also be unresponsive at times - so **please do not rely on them for validation**. We may also disable them at any time without warning.
```

File: docs/the_nimbus_book/src/rest-api.md [hunk 4]
```diff
@@ -1,31 +1,172 @@
...
+
+
+## Configure your node to run a local REST server
+
+> **Warning:** If you are using a validator client with a Nimbus beacon node, and running a Nimbus version prior to `v1.5.5`,  then you will need to launch the node with the `--subscribe-all-subnets` option enabled (in addition to the `--rest` option).
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/rest-api.md]: let's move this comment to the end of the section, it becomes less and less relevant with every release

File: docs/the_nimbus_book/src/rest-api.md [hunk 5]
```diff
@@ -1,31 +1,172 @@
...
+
+By default, the REST interface is disabled. To enable it, start the beacon node with the `--rest` option:
+
+```
+./run-prater-beacon-node.sh --rest
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/rest-api.md]: we have a mix of mainnet / prater throughout the manual - we should settle on one I think (mainnet probably).

Comment (arnetheduck) [reply docs/the_nimbus_book/src/rest-api.md]: this helps establish 5052 as the "mainnet beacon node port" - then we can use 5152 (or something) in examples that involve testnets

**Review by arnetheduck — APPROVED** [review]

File: docs/the_nimbus_book/src/rest-api.md (hunk 5 above)
Comment (unixpi) [reply docs/the_nimbus_book/src/rest-api.md]: sounds good

**Pull request merged**